--- vpnc-script-win.js	2023-12-01 09:25:46.608424600 +0200
+++ vpnc-script.js	2023-12-21 23:05:12.399446200 +0200
@@ -52,13 +52,20 @@
 
 function echo(level, msg)
 {
+    var msg_write;
     if (logLevel < level)
         return;
 
     if (logTimestamps)
-        WScript.echo("[" + ocTimestamp(new Date()) + "] " + msg);
+        msg_write = "[" + ocTimestamp(new Date()) + "] " + msg;
     else
-        WScript.echo(msg);
+        msg_write = msg;
+        
+    if (env("LOG2FILE")) {
+        log.WriteLine(msg_write);
+    } else {
+        WScript.echo(msg_write);
+    }
 }
 
 function run(cmd)
@@ -97,14 +104,20 @@
 // Script starts here
 // --------------------------------------------------------------
 
+if (env("LOG2FILE")) {
+	var fs = WScript.CreateObject("Scripting.FileSystemObject");
+	var tmpdir = fs.GetSpecialFolder(2)+"\\";
+	var log = fs.OpenTextFile(tmpdir + "vpnc.log", 8, true);
+}
+
 switch (env("reason")) {
 case "pre-init":
     break;
 case "connect":
     if (env("CISCO_BANNER")) {
-        echo(INFO, "--------------------------------------------------");
+        echo(INFO, "--------------------- BANNER ---------------------");
         echo(INFO, env("CISCO_BANNER"));
-        echo(INFO, "--------------------------------------------------");
+        echo(INFO, "------------------- BANNER end -------------------");
     }
 
     var gw = getDefaultGateway();
@@ -260,6 +273,7 @@
     echo(INFO, "Removing explicit route to VPN gateway " + env("VPNGATEWAY"));
     run("route delete " + env("VPNGATEWAY") + " mask 255.255.255.255");
 
+
     // Delete address
     echo(INFO, "Removing" + (env("INTERNAL_IP6_ADDRESS") ? " IPv6 and" : "") + " Legacy IP addresses");
     run("netsh interface ipv4 delete address " + env("TUNIDX") + " " +
@@ -282,4 +296,9 @@
     // FIXME: handle IPv6 split-excludes
     echo(INFO, "done.");
 }
+
+if (env("LOG2FILE")) {
+	log.Close();
+}
+
 WScript.Quit(accumulatedExitCode);
